e2e/create-cluster_test.go

```diff
@@ -112,23 +111,20 @@ var _ = Describe("CreateCluster", func() {
 	})
 
 	waitForBootstrappedMachines := func() {
-		Eventually(func() error {
+		Eventually(func(g Gomega) {
 			machineList := &infrav1.KubevirtMachineList{}
-			err := k8sclient.List(context.Background(), machineList, client.InNamespace(namespace))
-			if err != nil {
-				return err
-			}
+			g.Expect(
+				k8sclient.List(context.Background(), machineList, client.InNamespace(namespace)),
+			).To(Succeed())
 
-			if len(machineList.Items) == 0 {
-				return fmt.Errorf("expecting a non-empty list of machines")
-			}
+			g.Expect(machineList.Items).ToNot(BeEmpty(), "expecting a non-empty list of machines")
 
 			for _, machine := range machineList.Items {
-				if !conditions.IsTrue(&machine, infrav1.BootstrapExecSucceededCondition) {
-					return fmt.Errorf("still waiting on a kubevirt machine with bootstrap succeeded condition")
-				}
+				g.Expect(conditions.IsTrue(&machine, infrav1.BootstrapExecSucceededCondition)).To(
+					BeTrue(),
+					"still waiting on a kubevirt machine with bootstrap succeeded condition",
+				)
 			}
```

Comment:
We're waiting to "eventually" have this loop pass without errors. By introducing the g.Expect logic in this loop, will the eventually loop still be eventually consistent once no errors are encountered?

Reply:
Yes, exactly. See here https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode

Reply:
oh perfect. okay